tests/smoke_test.rs

```diff
@@ -0,0 +1,125 @@
+// Copyright 2022 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::path::Path;
+
+use anyhow::Result as AnyResult;
+use insta::assert_yaml_snapshot;
+use opa_wasm::{read_bundle, Runtime};
+use serde_json::json;
+use wasmtime::{Config, Engine, Module, Store};
+
+macro_rules! integration_test {
+    ($name:ident, $suite:expr) => {
+        #[tokio::test]
+        async fn $name() {
+            assert_yaml_snapshot!(test_policy($suite, None)
+                .await
+                .expect("error in test suite"));
+        }
+    };
+    ($name:ident, $suite:expr, input = $input:expr) => {
+        #[tokio::test]
+        async fn $name() {
+            assert_yaml_snapshot!(test_policy($suite, Some($input))
+                .await
+                .expect("error in test suite"));
+        }
+    };
+}
+
+async fn eval_policy(
+    bundle: &str,
+    entrypoint: &str,
+    input: &serde_json::Value,
+) -> AnyResult<serde_json::Value> {
+    let module = read_bundle(bundle).await?;
+
+    // Configure the WASM runtime
+    let mut config = Config::new();
+    config.async_support(true);
+
+    let engine = Engine::new(&config)?;
+
+    let module = Module::new(&engine, module)?;
+
+    // Create a store which will hold the module instance
+    let mut store = Store::new(&engine, ());
+
+    let data = serde_json::json!({});
+
+    // Instantiate the module
+    let runtime = Runtime::new(&mut store, &module).await?;
+
+    let policy = runtime.with_data(&mut store, &data).await?;
```

Comment:
I guess you could call `without_data` here instead

Reply:
done